### PR TITLE
[Docs]: Template Documents

### PR DESCRIPTION
**Pull request by gluax**
Adds our default PR, and Issue templates. The issue templates have the labels changed to match the labels we discussed in our meeting as well.

Adds the basic README and CONTRIBUTING files that are generic enough to suit all our projects.

Many files refer to `the-repo-name`. Simply find and replace all instances of that with the corresponding repo name.

I also made some updates that @Thomasvdam suggested in this [PR](https://github.com/sedaprotocol/seda-chain/pull/4).
As well as some other minor fixes.
Lastly adds references to SIPs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+### Change Size
+
+Please try to keep changes small to make reviews easier. We will reject more
+extensive unless there is a valid reason.
+
+### Formatting and Cleanliness
+
+Please ensure your code is formatted and clippy gives no warnings.
+
```
Comment (Thomasvdam) [inline CONTRIBUTING.md]: The clippy comment is Rust only.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Commit messages should always be meaningful.
+
+## Getting Ready For a PR
+
+This section describes actions to keep in mind while developing.
+
+### Change Size
+
+Please try to keep changes small to make reviews easier. We will reject more
+extensive unless there is a valid reason.
```
Comment (Thomasvdam) [inline CONTRIBUTING.md]: Perhaps something like "Unless there is no way to break up the changes we will reject large PRs"

**Review by Thomasvdam — APPROVED** [review]

**Review by jamesondh — APPROVED** [review]
Does it make sense to include a template for chore and/or refactor too?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+   to the `main` branch of `the-repo-name`.
+2. Please fill in the PR template that is there.
+3. Then assign it to yourself and anyone else who worked on the issue with you.
+4. Make sure all CI tests pass.
+5. Finally, please assign at least two of the following reviewers to your PR:
+   - [FranklinWaller](https://github.com/FranklinWaller)
+   - [gluax](https://github.com/gluax)
+   - [jamesondh](https://github.com/jamesondh)
+   - [mariocao](https://github.com/mariocao)
+   - [mennatabuelnaga](https://github.com/mennatabuelnaga)
```
[the remark this answers is not in the document]
Comment (gluax) [reply CONTRIBUTING.md:58]: Oh Sorry!!! I'll add you haha

**Comment by gluax** [comment]
@jamesondh I thought about that! I can certainly add one. I think it makes sense to make it more of a pr template than an Issue one.

**Comment by gluax** [comment]
Actually not, sure if we can have multiple PR templates. But, I will look into it.

If not, do people think we need a refactor or chore issue template? Maybe a refactor request could be a good place for a user to recommend ergonomic/ux friendly changes rather than a full-on feature request.

**Comment by Thomasvdam** [comment]
I think you can only have one PR template. Jameson's suggestion makes sense, we usually know whether it's a refactor or chore when we're able to describe the task in an issue. :)

**Comment by gluax** [comment]
Added the chore and refactor issue templates.

Added emojis, that much our [labels](https://github.com/sedaprotocol/seda-chain/labels) when appropriate.

I made suggested changes. Added a `.gitignore` so local editor files don't get added to the repo. But this does not get copied over to the repos made with this template, from my understanding.

**Review by jamesondh — APPROVED** [review]

**Review by Thomasvdam — APPROVED** [review]

**Review by mariocao — APPROVED** [review]
I think that the `README.md` is [not supported](https://docs.github.com/en/enterprise-server@3.6/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file) as default for our org but it can be used as an example for future repos.

**Pull request merged**